tests/fail/leak_in_tls.rs

```diff
@@ -0,0 +1,24 @@
+//@error-in-other-file: memory leaked
+//@normalize-stderr-test: ".*│.*" -> "$$stripped$$"
+
+use std::cell::Cell;
+
+pub fn main() {
+    thread_local! {
+        static REF: Cell<Option<&'static i32>> = Cell::new(None);
+    }
```

Comment:
So this will test that we allow leaks in one of the 2 kinds of TLS (TLS statics vs OS-supported TLS keys). We should make sure to have tests covering both (by directly using the underlying mechanisms).

Reply:
Added `#[thread_local]`s to the tests

Reply:
I think we want one test that uses `#[thread_local]` and one that uses pthread TLS keys directly.

Or we only treat `#[thread_local]` as static roots? Then we cover all targets with `target_thread_local`. Doing special magic to library stuff like the pthread keys is somewhat dubious anyway...